### PR TITLE
fix(TBD-13144): fix for tKafkaInput on spark 2.4x

### PR DESCRIPTION
**Pull request by sponomarova**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.spark.distribution.spark24x/plugin.xml [hunk 1]
```diff
@@ -380,8 +380,8 @@
         <!-- Spark Streaming Kafka group -->
         <libraryNeededGroup description="Spark streaming Kafka libraries for spark247" id="KAFKA-GROUP-SPARK_2_4_x" name="KAFKA-GROUP-SPARK_2_4_x">
             <library id="kafka-clients-2.7.0.jar"/>
-            <library id="kafka_2.12-2.7.0.jar"/>
-            <library id="spark-streaming-kafka-0-10_2.12-2.4.7.jar"/>
+            <library id="kafka_2.11-2.4.0.jar"/>
```
Comment (AlixMetivier) [inline main/plugins/org.talend.spark.distribution.spark24x/plugin.xml:383]: is it alright to mix kafka client and spark streaming kafka 2.7.0 with kafka 2.4.0 ?

**Review by cbadillo1603 — DISMISSED** [review]
reviewed

File: main/plugins/org.talend.spark.distribution.spark24x/plugin.xml (hunk 1 above)
Comment (sponomarova) [reply main/plugins/org.talend.spark.distribution.spark24x/plugin.xml:383]: Agree. it is not good. 
To have the same versions I will have to put kafka-clients of 2.4.0 to have compatibility with scala. 
For spark 24x we are using scala 2.11. According to this: https://mvnrepository.com/artifact/org.apache.kafka/kafka
**kafka_2.11-2.4.0.jar** is the most recent lib based on scala 2.11.

File: main/plugins/org.talend.spark.distribution.spark24x/plugin.xml [hunk 2]
```diff
@@ -379,9 +379,9 @@
 
         <!-- Spark Streaming Kafka group -->
         <libraryNeededGroup description="Spark streaming Kafka libraries for spark247" id="KAFKA-GROUP-SPARK_2_4_x" name="KAFKA-GROUP-SPARK_2_4_x">
-            <library id="kafka-clients-2.7.0.jar"/>
-            <library id="kafka_2.12-2.7.0.jar"/>
-            <library id="spark-streaming-kafka-0-10_2.12-2.4.7.jar"/>
+            <library id="kafka-clients-2.4.0.jar"/>
+            <library id="kafka_2.11-2.4.0.jar"/>
+            <library id="spark-streaming-kafka-0-10_2.11-2.4.7.jar"/>
```
Comment (AlixMetivier) [inline main/plugins/org.talend.spark.distribution.spark24x/plugin.xml:384]: this one should also be 2.4.0 ?

**Review by cbadillo1603 — APPROVED** [review]
reviewed

**Pull request merged**